### PR TITLE
[Network] az network dns record-set: Support TMLink recordset provisioning

### PR DESCRIPTION
**Pull request by vatsalyachauhan**
Corresponding az-cli PR: [Azure/azure-cli#28516](https://github.com/Azure/azure-cli/pull/28516)

### CODE REVIEW

File: Resources/mgmt-plane/L3N1YnNjcmlwdGlvbnMve30vcmVzb3VyY2Vncm91cHMve30vcHJvdmlkZXJzL21pY3Jvc29mdC5uZXR3b3JrL2Ruc3pvbmVzL3t9L3t9L3t9/2023-07-01-preview.xml [hunk 1]
```diff
@@ -829,6 +837,9 @@
         <arg nullable="True" type="string" var="$parameters.properties.targetResource.id" options="target-resource" group="">
           <help short="ID of an Azure resource from which the DNS resource value is taken."/>
         </arg>
+        <arg nullable="True" type="string" var="$parameters.properties.trafficManagementProfile.id" options="traffic-management-profile" stage="Experimental" group="">
+          <help short="A reference to an azure traffic manager profile resource from where the dns resource value is taken."/>
```
Comment (djr-jsr) [inline Resources/mgmt-plane/L3N1YnNjcmlwdGlvbnMve30vcmVzb3VyY2Vncm91cHMve30vcHJvdmlkZXJzL21pY3Jvc29mdC5uZXR3b3JrL2Ruc3pvbmVzL3t9L3t9L3t9/2023-07-01-preview.xml:841]: Capitalize Azure Traffic Manager and DNS

File: Resources/mgmt-plane/L3N1YnNjcmlwdGlvbnMve30vcmVzb3VyY2Vncm91cHMve30vcHJvdmlkZXJzL21pY3Jvc29mdC5uZXR3b3JrL2Ruc3pvbmVzL3t9L3t9L3t9/2023-07-01-preview.xml [hunk 2]
```diff
@@ -318,6 +319,9 @@
         <arg type="string" var="$parameters.properties.targetResource.id" options="target-resource" group="">
           <help short="ID of an Azure resource from which the DNS resource value is taken."/>
         </arg>
+        <arg type="string" var="$parameters.properties.trafficManagementProfile.id" options="traffic-management-profile" stage="Experimental" group="">
+          <help short="A reference to an azure traffic manager profile resource from where the dns resource value is taken."/>
```
Comment (djr-jsr) [inline Resources/mgmt-plane/L3N1YnNjcmlwdGlvbnMve30vcmVzb3VyY2Vncm91cHMve30vcHJvdmlkZXJzL21pY3Jvc29mdC5uZXR3b3JrL2Ruc3pvbmVzL3t9L3t9L3t9/2023-07-01-preview.xml:323]: Capitalize Azure Traffic Manager and DNS

**Review by djr-jsr — CHANGES_REQUESTED** [review]

File: Resources/mgmt-plane/L3N1YnNjcmlwdGlvbnMve30vcmVzb3VyY2Vncm91cHMve30vcHJvdmlkZXJzL21pY3Jvc29mdC5uZXR3b3JrL2Ruc3pvbmVzL3t9L3t9L3t9/2023-07-01-preview.xml (hunk 1 above)
Comment (vatsalyachauhan) [reply Resources/mgmt-plane/L3N1YnNjcmlwdGlvbnMve30vcmVzb3VyY2Vncm91cHMve30vcHJvdmlkZXJzL21pY3Jvc29mdC5uZXR3b3JrL2Ruc3pvbmVzL3t9L3t9L3t9/2023-07-01-preview.xml]: Updated.

File: Resources/mgmt-plane/L3N1YnNjcmlwdGlvbnMve30vcmVzb3VyY2Vncm91cHMve30vcHJvdmlkZXJzL21pY3Jvc29mdC5uZXR3b3JrL2Ruc3pvbmVzL3t9L3t9L3t9/2023-07-01-preview.xml (hunk 2 above)
Comment (vatsalyachauhan) [reply Resources/mgmt-plane/L3N1YnNjcmlwdGlvbnMve30vcmVzb3VyY2Vncm91cHMve30vcHJvdmlkZXJzL21pY3Jvc29mdC5uZXR3b3JrL2Ruc3pvbmVzL3t9L3t9L3t9/2023-07-01-preview.xml]: Updated.

**Review by djr-jsr — APPROVED** [review]

**Review by djr-jsr — APPROVED** [review]

**Review by djr-jsr — APPROVED** [review]

**Review by necusjz — APPROVED** [review]

**Pull request merged**